internal/observability/middleware_test.go

```diff
@@ -170,7 +183,10 @@ func TestMiddlewareLogging(t *testing.T) {
 			assert.Equal(t, expected, getLog(), "Unexpected log entry written.")
 		})
 		t.Run(tt.desc+", oneway inbound", func(t *testing.T) {
-			err := mw.HandleOneway(context.Background(), req, fakeHandler{tt.err, false})
+			if tt.applicationErr {
+				t.Skip()
+			}
```

Comment:
Let's pull this out of the sub-tests and reduce some of the repetition - we can throw a `continue` into the loop instead of skipping each test.

```
for _, tt := range tests {
  ...run the unary sub-tests...
  
  // Application errors aren't applicable to streaming and oneway.
  if tt.applicationError {
    continue
  }
  ...remaining sub-tests...
}
```